docs/functionaldatatypes/nonemptylist.md

```diff
@@ -0,0 +1,195 @@
+---
+id: nonemptylist
+title: "NonEmptyList"
+---
+
+`NonEmptyList` represents a `List` that is guaranteed not to be empty.
+
+This is useful to allow more accurate domain modeling in situations where we know a collection is not empty.
+
+For example, we might be handling a collection of errors.
+
+If an error has occurred at all then we know the collection is not empty. However, the Scala compiler does not know that and so we are forced to handle the case where the collection is empty even though we know that cannot occur.
```

Comment:
sugg: the Scala compiler does not know that in general for the standard library collections such as `Iterable`, `List`, `Vector`.
My point really is to be a bit more concrete about what Scala compiler does not know by clarifying what collections we have in mind. The actual wording I propose is not particularly important but illustrates the point.